auslib/db.py

```diff
@@ -1008,6 +1012,23 @@ def _dataVersionsAreSynced(self, sc_id, transaction):
 
         return True
 
+    def _alreadyCompleted(self, where, transaction, change):
+        """
+        Method to verify whether the scheduled change has already been completed or not. If completed,
+        then cannot modify the scheduled change anymore.
+
+        @param where: A list of SQLAlchemy clauses, or a key/value pair of columns and values.
+        @type where: list of clauses or key/value pairs.
+
+        @param transaction: A transaction object to add the update statement (and history changes) to.
+                           If provided, you must commit the transaction yourself. If None, they will
+                           be added to a locally-scoped transaction and committed.
+        @param change: a string denoting the type of change. Eg: Delete/Update.
+        """
+        completed_scheduled_change = self.select(where=where, transaction=transaction, columns=["complete"])
```

Comment:
This method isn't really doing much, and raising the exception here makes things a bit harder to read. I think you should either be raising the exception in the caller, or integrating this with .validate().